proto/rill/ui/v1/dashboard.proto

```diff
@@ -72,62 +72,58 @@ message DashboardState {
+  repeated DashboardDimensionFilter having = 4;
   // Selected time granularity
-  rill.runtime.v1.TimeGrain time_grain = 3;
+  rill.runtime.v1.TimeGrain time_grain = 5;
 
-  DashboardTimeRange compare_time_range = 4;
+  DashboardTimeRange compare_time_range = 6;
 
   // Determines the active page. This is cleaner than using setting `selected_dimension` or `expanded_measure`
-  ActivePage active_page = 32;
+  ActivePage active_page = 7;
 
   // Selected measure for the leaderboard
-  optional string leaderboard_measure = 5;
+  optional string leaderboard_measure = 8;
   // Focused dimension
-  optional string selected_dimension = 6;
+  optional string selected_dimension = 9;
 
-  optional bool show_time_comparison = 7;
+  optional bool show_time_comparison = 10;
 
   // Selected measures and dimensions to be shown
-  repeated string visible_measures = 8;
-  optional bool all_measures_visible = 9;
-  repeated string visible_dimensions = 10;
-  optional bool all_dimensions_visible = 11;
+  repeated string visible_measures = 11;
+  optional bool all_measures_visible = 12;
+  repeated string visible_dimensions = 13;
+  optional bool all_dimensions_visible = 14;
 
-  optional LeaderboardContextColumn leaderboard_context_column = 12;
+  optional LeaderboardContextColumn leaderboard_context_column = 15;
 
   // Selected timezone for the dashboard
-  optional string selected_timezone = 13;
+  optional string selected_timezone = 16;
 
   // Scrub time range
-  optional DashboardTimeRange scrub_range = 14;
+  optional DashboardTimeRange scrub_range = 17;
 
-  optional LeaderboardSortDirection leaderboard_sort_direction = 15;
-  optional LeaderboardSortType leaderboard_sort_type = 16;
+  optional LeaderboardSortDirection leaderboard_sort_direction = 18;
+  optional LeaderboardSortType leaderboard_sort_type = 19;
 
-  optional string comparison_dimension = 17;
+  optional string comparison_dimension = 20;
 
   // Expanded measure for TDD view
-  optional string expanded_measure = 18;
+  optional string expanded_measure = 21;
   // Pin index for TDD table selected values
-  optional int32 pin_index = 19;
+  optional int32 pin_index = 22;
   // Type of visualization for TDD view
-  optional string chart_type = 33;
+  optional string chart_type = 23;
 
   /**
    * Pivot related fields
    */
-  optional bool pivot_is_active = 22;
-  // List of time dimensions selected for rows
-  repeated rill.runtime.v1.TimeGrain pivot_row_time_dimensions = 23;
+  optional bool pivot_is_active = 24;
   // List of dimensions selected for rows
-  repeated string pivot_row_dimensions = 24;
-  // List of time dimensions selected for columns
-  repeated rill.runtime.v1.TimeGrain pivot_column_time_dimensions = 25;
-  // List of time dimensions selected for columns
-  repeated string pivot_column_dimensions = 26;
+  repeated PivotElement pivot_row_dimensions = 25;
+  // List dimensions selected for columns
+  repeated PivotElement pivot_column_dimensions = 26;
```

Comment:
We should add these at the end index and add backwards compatibility for previous fields